crates/loro-internal/src/encoding/encode_reordered.rs

```diff
@@ -1559,7 +1565,12 @@ mod value {
                     ValueKind::Binary
                 }
                 LoroValue::Container(c) => {
-                    let idx = register_cid.register(c);
+                    let idx = if register_cid.contains(c) {
+                        register_cid.register(c)
+                    } else {
+                        let container_type_u8 = c.container_type().to_u8();
+                        register_cid.len() + container_type_u8 as usize
```

Comment:
Maybe it's better to use negative value to represent container_type. Otherwise, if we already need 3 bytes to represent `register_cid.len()`, we need 3 bytes to represent each `container_type` also. But `container_type` should only take 1 byte

Reply:
Can I assume that only child container creation ops need to covert `ContainerIdx` to `ContainerID`？  Maybe we could only encode `ContainerType` by `u8` and not worry about whether it's in the `cids`.   I have tested it by fuzzing, it can pass.